nova/scheduler/external.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2025 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+This module provides functionality to interact with an external scheduler API
+to reorder and filter hosts based on additional criteria.
+
+The external scheduler API is expected to take a list of weighed hosts and
+their weights, along with the request specification, and return a reordered
+and filtered list of host names.
+"""
+from oslo_log import log as logging
+import requests
+
+from nova.api.validation import validators
```

Comment:
This is something I don't really like. Would it be hard to use the json schema library without importing from `nova.api`? My pain point here is, that `nova.scheduler` should from an architecture perspective not rely on code destined to be used in `nova-api`.

It doesn't look too complicated, as I don't think we need the extensions `nova.api` adds onto the `jsonschema` validator.

Reply:
Right, I'll use the library directly.